PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:14 + add:0, change matches:
-    _inherit = 'account.move'
-    _inherit = 'account.move.line'
-    _inherit = "account.tax"
-    _inherit = 'account.analytic.account'
-    _inherit = 'account.analytic.applicability'
-    _inherit = 'ir.actions.report'
-    _inherit = 'product.template'
-    _inherit = 'product.product'
-    _inherit = "product.supplierinfo"
-    _inherit = 'product.packaging'
-    _inherit = 'analytic.mixin'
-    _inherit = 'res.company'
-    _inherit = 'res.config.settings'
-    _inherit = 'res.partner'

Total Changes: 46

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/purchase/models/account_invoice.py
+++ b/addons/purchase/models/account_invoice.py
@@ -13,7 +13,7 @@ TOLERANCE = 0.02  # tolerance applied to the total when searching for a matching
 
 
 class AccountMove(models.Model):
-    _inherit = 'account.move'
+    _inherit = ['account.move']
 
     purchase_vendor_bill_id = fields.Many2one('purchase.bill.union', store=False, readonly=False,
         string='Auto-complete',
@@ -508,10 +508,9 @@ class AccountMove(models.Model):
                     })]
 
 
-
 class AccountMoveLine(models.Model):
     """ Override AccountInvoice_line to add the link to the purchase order line it is related to"""
-    _inherit = 'account.move.line'
+    _inherit = ['account.move.line']
 
     is_downpayment = fields.Boolean()
     purchase_line_id = fields.Many2one('purchase.order.line', 'Purchase Order Line', ondelete='set null', index='btree_not_null', copy=False)

--- a/addons/purchase/models/account_tax.py
+++ b/addons/purchase/models/account_tax.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class AccountTax(models.Model):
-    _inherit = "account.tax"
+    _inherit = ["account.tax"]
 
     def _hook_compute_is_used(self, taxes_to_compute):
         # OVERRIDE in order to fetch taxes used in purchase

--- a/addons/purchase/models/analytic_account.py
+++ b/addons/purchase/models/analytic_account.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models, _
 
 
 class AccountAnalyticAccount(models.Model):
-    _inherit = 'account.analytic.account'
+    _inherit = ['account.analytic.account']
 
     purchase_order_count = fields.Integer("Purchase Order Count", compute='_compute_purchase_order_count')
 

--- a/addons/purchase/models/analytic_applicability.py
+++ b/addons/purchase/models/analytic_applicability.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class AccountAnalyticApplicability(models.Model):
-    _inherit = 'account.analytic.applicability'
+    _inherit = ['account.analytic.applicability']
     _description = "Analytic Plan's Applicabilities"
 
     business_domain = fields.Selection(

--- a/addons/purchase/models/ir_actions_report.py
+++ b/addons/purchase/models/ir_actions_report.py
@@ -5,7 +5,7 @@ from odoo.tools.pdf import OdooPdfFileReader, OdooPdfFileWriter
 
 
 class IrActionsReport(models.Model):
-    _inherit = 'ir.actions.report'
+    _inherit = ['ir.actions.report']
 
     def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
         # EXTENDS base

--- a/addons/purchase/models/product.py
+++ b/addons/purchase/models/product.py
@@ -9,8 +9,7 @@ from dateutil.relativedelta import relativedelta
 
 
 class ProductTemplate(models.Model):
-    _name = 'product.template'
-    _inherit = 'product.template'
+    _inherit = ['product.template']
 
     purchased_product_qty = fields.Float(compute='_compute_purchased_product_qty', string='Purchased', digits='Product Unit of Measure')
     purchase_method = fields.Selection([
@@ -56,8 +55,7 @@ class ProductTemplate(models.Model):
 
 
 class ProductProduct(models.Model):
-    _name = 'product.product'
-    _inherit = 'product.product'
+    _inherit = ['product.product']
 
     purchased_product_qty = fields.Float(compute='_compute_purchased_product_qty', string='Purchased',
         digits='Product Unit of Measure')
@@ -117,7 +115,7 @@ class ProductProduct(models.Model):
 
 
 class ProductSupplierinfo(models.Model):
-    _inherit = "product.supplierinfo"
+    _inherit = ["product.supplierinfo"]
 
     @api.onchange('partner_id')
     def _onchange_partner_id(self):
@@ -125,6 +123,6 @@ class ProductSupplierinfo(models.Model):
 
 
 class ProductPackaging(models.Model):
-    _inherit = 'product.packaging'
+    _inherit = ['product.packaging']
 
     purchase = fields.Boolean("Purchase", default=True, help="If true, the packaging can be used for purchase orders")

--- a/addons/purchase/models/purchase_bill_line_match.py
+++ b/addons/purchase/models/purchase_bill_line_match.py
@@ -6,8 +6,7 @@ from odoo.tools import SQL
 from odoo.exceptions import UserError
 
 
-class PurchaseBillMatch(models.Model):
-    _name = "purchase.bill.line.match"
+class PurchaseBillLineMatch(models.Model):
     _description = "Purchase Line and Vendor Bill line matching view"
     _auto = False
     _order = 'product_id, aml_id, pol_id'

--- a/addons/purchase/models/purchase_order.py
+++ b/addons/purchase/models/purchase_order.py
@@ -16,7 +16,6 @@ from odoo.exceptions import UserError, ValidationError
 
 
 class PurchaseOrder(models.Model):
-    _name = "purchase.order"
     _inherit = ['portal.mixin', 'product.catalog.mixin', 'mail.thread', 'mail.activity.mixin']
     _description = "Purchase Order"
     _rec_names_search = ['name', 'partner_ref']

--- a/addons/purchase/models/purchase_order_line.py
+++ b/addons/purchase/models/purchase_order_line.py
@@ -10,8 +10,7 @@ from odoo.exceptions import UserError
 
 
 class PurchaseOrderLine(models.Model):
-    _name = 'purchase.order.line'
-    _inherit = 'analytic.mixin'
+    _inherit = ['analytic.mixin']
     _description = 'Purchase Order Line'
     _order = 'order_id, sequence, id'
 

--- a/addons/purchase/models/res_company.py
+++ b/addons/purchase/models/res_company.py
@@ -3,8 +3,9 @@
 
 from odoo import fields, models
 
-class Company(models.Model):
-    _inherit = 'res.company'
+
+class ResCompany(models.Model):
+    _inherit = ['res.company']
 
     po_lead = fields.Float(string='Purchase Lead Time', required=True,
         help="Margin of error for vendor lead times. When the system "

--- a/addons/purchase/models/res_config_settings.py
+++ b/addons/purchase/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
+    _inherit = ['res.config.settings']
 
     lock_confirmed_po = fields.Boolean("Lock Confirmed Orders", default=lambda self: self.env.company.po_lock == 'lock')
     po_lock = fields.Selection(related='company_id.po_lock', string="Purchase Order Modification *", readonly=False)

--- a/addons/purchase/models/res_partner.py
+++ b/addons/purchase/models/res_partner.py
@@ -1,12 +1,11 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo import api, fields, models, _
-from odoo.addons.base.models.res_partner import WARNING_MESSAGE, WARNING_HELP
+from odoo.addons.base.models.ResPartner import WARNING_MESSAGE, WARNING_HELP
 
 
-class res_partner(models.Model):
-    _name = 'res.partner'
-    _inherit = 'res.partner'
+class ResPartner(models.Model):
+    _inherit = ['res.partner']
 
     def _compute_purchase_order_count(self):
         self.purchase_order_count = 0
@@ -53,7 +52,7 @@ class res_partner(models.Model):
 
     @api.model
     def _commercial_fields(self):
-        return super(res_partner, self)._commercial_fields()
+        return super()._commercial_fields()
 
     property_purchase_currency_id = fields.Many2one(
         'res.currency', string="Supplier Currency", company_dependent=True,
